From 5abc51dc5585ce9cc97bfabb9f09a7b6645ccd6f Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Mon, 5 Jul 2021 03:08:48 -0700
Subject: [PATCH 1626/1921] octeontx2-af: remove NPA resource limits check

The SDP PF/VF do not need NPA LFs as they are not used. By reserving the
LF count here the actual NPA PF device is not getting any LF resources
and the PF driver is failing in resource policy check. In case of huge
number of SDP VF case this becomes a problem and making the system
unusable. This patch fixes such situation by removing the NPA resource
altogether for anyone.

Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Change-Id: Ia0caaa12f7cd97ca3ac94319d10de477ac46259e
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/55676
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Satananda Burla <sburla@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 .../ethernet/marvell/octeontx2/af/rvu_validation.c  | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
index f71c80edfeff..b64d4058e117 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
@@ -517,17 +517,6 @@ int rvu_check_rsrc_policy(struct rvu *rvu, struct rsrc_attach *req,
 	if (!is_rvu_otx2(rvu))
 		return 0;
 
-	/* Only one NPA LF can be attached */
-	if (req->npalf) {
-		block = &hw->block[BLKADDR_NPA];
-		free_lfs = rvu_rsrc_free_count(&block->lf);
-		limit = rvu->pf_limits.npa->a[pf].val;
-		familylfs = rvu_blk_count_rsrc(block, pcifunc,
-					       RVU_PFVF_PF_SHIFT);
-		if (!free_lfs || (limit == familylfs))
-			goto fail;
-	}
-
 	/* Only one NIX LF can be attached */
 	if (req->nixlf) {
 		block = &hw->block[BLKADDR_NIX0];
@@ -705,9 +694,7 @@ static void rvu_set_default_limits(struct rvu *rvu)
 			break;
 		case PCI_DEVID_OCTEONTX2_SDP_RVU_PF:
 			nnix -= 1 + nvfs;
-			nnpa -= 1 + nvfs;
 			rvu->pf_limits.nix->a[i].val = nnix > 0 ? 1 + nvfs : 0;
-			rvu->pf_limits.npa->a[i].val = nnpa > 0 ? 1 + nvfs : 0;
 			if (rvu->hw->cap.nix_fixed_txschq_mapping)
 				break;
 			rvu->pf_limits.smq->a[i].val = nsmq;
-- 
2.31.1

